### ISSUE TITLE
Support Read Replicas

### ISSUE THREAD

**Issue by jmymy** [terraform-google-modules/terraform-google-memorystore#83]
### TL;DR

Drift when using latest version of the provider that tries to update the field read_replicas_mode. This needs to be supported in this module, otherwise the instance is re-created.

### Expected behavior

Be able to configure read_replicas.

### Observed behavior

`      + read_replicas_mode       = "READ_REPLICAS_DISABLED" # forces replacement`

### Terraform Configuration

```hcl
module "redis" {
  source                  = "terraform-google-modules/memorystore/google"
  version                 = "4.2.0"
  name                    = "redis"
  project                 = module.project.project_id
  authorized_network      = module.kubernetes.network_name
  memory_size_gb          = 1
  redis_version           = "REDIS_6_X"
  connect_mode            = "PRIVATE_SERVICE_ACCESS"
  auth_enabled            = true
  transit_encryption_mode = var.transit_encryption_mode
}
```

### Terraform Version

```sh
Terraform v1.1.8
on darwin_arm64
+ provider registry.terraform.io/hashicorp/external v2.2.2
+ provider registry.terraform.io/hashicorp/google v4.16.0
+ provider registry.terraform.io/hashicorp/google-beta v4.16.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.10.0
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.1.2
+ provider registry.terraform.io/hashicorp/tfe v0.30.2
+ provider registry.terraform.io/hashicorp/time v0.7.2
```

### Additional information

[read_replicas_mode](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/redis_instance#read_replicas_mode) - (Optional) Optional. Read replica mode. Can only be specified when trying to create the instance. If not set, Memorystore Redis backend will default to READ_REPLICAS_DISABLED.

**Comment by jmymy** [comment]
looks like they fixed the bug in https://github.com/hashicorp/terraform-provider-google-beta/pull/4184
